src/components/location-search/location-search.tsx

```diff
@@ -0,0 +1,68 @@
+import React, { useState } from "react";
+import { GeocodingResult, useGeocoding } from "./hooks/use-geocoding";
+import ClearIcon from "../icons/clear-icon";
+import SearchIcon from "../icons/search-icon";
+import { useMapStore } from "../map/map-store";
+
+const LocationSearch: React.FC = ({}) => {
+  const [search, setSearch] = useState("");
+  const geocodingResults = useGeocoding(search);
+  const hasResults = geocodingResults.length;
+  const { map } = useMapStore();
```

Comment:
are you sure this works? don't we need to use it like this with zustand stores:
`const map = useMapStore((state) => state.map);`

Reply:
i just checked and looks like it does 😮 TIL